agent/api/task/task.go

```diff
@@ -302,7 +302,13 @@ func TaskFromACS(acsTask *ecsacs.Task, envelope *ecsacs.PayloadMessage) (*Task,
 		task.StopSequenceNumber = *envelope.SeqNum
 	}
 
-	// Overrides the container command if it's set
+	// get service connect config value from ACS attachment
+	scConfig, err := ExtractSCConfigFromACSAttachment(acsTask)
+	if err != nil {
+		return nil, err
```

Comment:
Can we wrap the error message to indicate that we have failed to extract SC config? Otherwise it may be hard to relate the failure to the actual error.

Reply:
Please find the update in commit 94753650e02342a2c1be58645e8e187489efe452. Thanks!
```
// extract the service connect container name and service connect config value from the service connect attachment
    if scAttachment != nil {
        scConfig, err := ParseServiceConnectAttachment(scAttachment, acsTask.Containers, aws.StringValue(acsTask.NetworkMode))
	 if err != nil {
	     return nil, fmt.Errorf("error parsing service connect config value from the service connect attachment: %w", err)
	 }
	 task.ServiceConnectConfig = scConfig
 }
```